### PR TITLE
[iOS] Fix ListView measure bugs

### LINKED ISSUE #2131: [iOS][ListView] ItemTemplate content is incorrectly measured
## Current behavior

The Content of a ListViewItem may be incorrectly rendered if the ListView is added late in the visual tree.

## How to reproduce it (as minimally and precisely as possible)

https://dev.azure.com/nventive/Umbrella/_workitems/edit/167308

## Environment

Nuget Package:

Package Version(s):

Affected platform(s):

- [x] iOS
- [ ] Android
- [ ] WebAssembly
- [ ] WebAssembly renderers for Xamarin.Forms
- [ ] Windows
- [ ] Build tasks
- [ ] Solution Templates

Visual Studio:

- [ ] 2017 (version: )
- [ ] 2019 (version: )
- [ ] for Mac (version: )

Relevant plugins:

- [ ] Resharper (version: )

### LINKED ISSUE #2146: [iOs] Async update of ListView with ObservableCollection not working
## Current behavior
A ObservableCollection gets updated from an async function, but the changes are not populated to the Listview it is Bound to. This works on UWP, Droid and WASM.

## Expected behavior
The items are populated once they are Added to the ObservableCollection.

## How to reproduce it (as minimally and precisely as possible)
See this GitHub-Repo: https://github.com/TopperDEL/UnoTest01
The update happens in OnNavigatedTo from the MainPage. Remove the async Task.Delay to see it update correctly.

My assumption is that ObservableCollection does not raise ItemAdded/Changed-Events on iOs.

## Environment

Nuget Package: v2.0.5

Package Version(s):

Affected platform(s):

- [X] iOS
- [ ] Android
- [ ] WebAssembly
- [ ] WebAssembly renderers for Xamarin.Forms
- [ ] Windows
- [ ] Build tasks
- [ ] Solution Templates

Visual Studio:

- [ ] 2017 (version: )
- [ ] 2019 (version: )
- [X] for Mac (version: )

Relevant plugins:

- [ ] Resharper (version: )

### PR DESCRIPTION
**Pull request by davidjohnoliver**
GitHub Issue (If applicable): fixes #2131, fixes #2146

Fix bugs where iOS ListView fails to take extra available space when its desired dimensions change.

## PR Type

- Bugfix
What kind of change does this PR introduce?

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tested code with current [supported SDKs](https://github.com/unoplatform/uno/blob/master/README.md#uno-features)
- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [x] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] [Wasm UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md#running-the-webassembly-ui-tests-snapshots) are not showing unexpected any differences. Validate PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [ ] Updated the [Release Notes](https://github.com/unoplatform/uno/tree/master/doc/ReleaseNotes)
- [x] Associated with an issue (GitHub or internal)

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/SamplesApp/SamplesApp.UITests/Windows_UI_Xaml_Controls/ListViewTests/UnoSamples_Tests.ListView.cs [hunk 1]
```diff
@@ -119,5 +119,44 @@ public void Check_ListView_Swallows_Measure()
...
+		public void ListView_Weird_Measure_During_Arrange()
+		{
+			Run("UITests.Shared.Windows_UI_Xaml_Controls.ListView.ListView_Weird_Measure");
+
+			_app.WaitForText("StatusTextBlock", "Finished");
+
+			TakeScreenshot($"{nameof(ListView_Weird_Measure_During_Arrange)} after layout");
+
+			var heightStr = _app.GetText("HeightTextBlock");
+			var height = int.Parse(heightStr);
```
Comment (jeromelaban) [inline src/SamplesApp/SamplesApp.UITests/Windows_UI_Xaml_Controls/ListViewTests/UnoSamples_Tests.ListView.cs]: You can use `_app.Query(q => q.Marked("dfdldks").First().Rect.Height` as well.

File: src/SamplesApp/SamplesApp.UITests/Windows_UI_Xaml_Controls/ListViewTests/UnoSamples_Tests.ListView.cs [hunk 2]
```diff
@@ -119,5 +119,44 @@ public void Check_ListView_Swallows_Measure()
...
+
+		[Test]
+		[AutoRetry]
+		public void ListView_Weird_Measure_During_Arrange()
+		{
+			Run("UITests.Shared.Windows_UI_Xaml_Controls.ListView.ListView_Weird_Measure");
+
+			_app.WaitForText("StatusTextBlock", "Finished");
+
+			TakeScreenshot($"{nameof(ListView_Weird_Measure_During_Arrange)} after layout");
```
Comment (jeromelaban) [inline src/SamplesApp/SamplesApp.UITests/Windows_UI_Xaml_Controls/ListViewTests/UnoSamples_Tests.ListView.cs]: The name of the test is included in the `TakeScreenshot` pass

File: src/SamplesApp/SamplesApp.UITests/Windows_UI_Xaml_Controls/ListViewTests/UnoSamples_Tests.ListView.cs (hunk 1 above)
Comment (davidjohnoliver) [reply src/SamplesApp/SamplesApp.UITests/Windows_UI_Xaml_Controls/ListViewTests/UnoSamples_Tests.ListView.cs]: Good point!

On Android will it be in physical or logical pixels?

File: src/SamplesApp/UITests.Shared/Windows_UI_Xaml_Controls/ListView/ListView_Weird_Measure.xaml [hunk 3]
```diff
@@ -0,0 +1,50 @@
+﻿<UserControl x:Class="UITests.Shared.Windows_UI_Xaml_Controls.ListView.ListView_Weird_Measure"
```
[the remark this answers is not in the document]
Comment (davidjohnoliver) [reply src/SamplesApp/UITests.Shared/Windows_UI_Xaml_Controls/ListView/ListView_Weird_Measure.xaml]: This is the censored version :p

File: src/SamplesApp/SamplesApp.UITests/Windows_UI_Xaml_Controls/ListViewTests/UnoSamples_Tests.ListView.cs (hunk 2 above)
Comment (davidjohnoliver) [reply src/SamplesApp/SamplesApp.UITests/Windows_UI_Xaml_Controls/ListViewTests/UnoSamples_Tests.ListView.cs]: fixed

File: src/SamplesApp/SamplesApp.UITests/Windows_UI_Xaml_Controls/ListViewTests/UnoSamples_Tests.ListView.cs (hunk 1 above)
Comment (jeromelaban) [reply src/SamplesApp/SamplesApp.UITests/Windows_UI_Xaml_Controls/ListViewTests/UnoSamples_Tests.ListView.cs]: It should not matter. If you need something to be of a certain size, take one as a reference and compare.

**Pull request merged**